mobile/library/cc/engine_builder.cc

```diff
@@ -747,6 +753,12 @@ std::unique_ptr<envoy::config::bootstrap::v3::Bootstrap> EngineBuilder::generate
         ->mutable_idle_network_timeout()
         ->set_seconds(quic_connection_idle_timeout_seconds_);
 
+    auto* quic_protocol_options = alpn_options.mutable_auto_config()
+                                      ->mutable_http3_protocol_options()
+                                      ->mutable_quic_protocol_options();
+    quic_protocol_options->mutable_connection_keepalive()->mutable_initial_interval()->set_nanos(
+        keepalive_initial_interval_ms_ * 1000 * 1000);
```

Comment:
since the default value is 0 in EngineBuilder, this will disable keep alive entirely by default: https://github.com/envoyproxy/envoy/blob/982c6bd6a74d3f0918c05b2c512f971240228d82/api/envoy/config/core/v3/protocol.proto#L51.

Reply:
Actually, it looks like the default is to disable keep alive anyway: https://github.com/envoyproxy/envoy/blob/982c6bd6a74d3f0918c05b2c512f971240228d82/source/common/quic/envoy_quic_client_session.cc#L258.

I thought we had keep alive but it was some larger value like 15s?

Reply:
Your first link:
```For a client connection, if :ref:`max_interval <envoy_v3_api_field_config.core.v3.QuicKeepAliveSettings.max_interval>`  is also zero, do not keepalive, otherwise use max_interval or QUICHE default to probe all the time. ```

Reply:
ah yes, sorry I missed that, thanks!

Reply:
so this appears to pass but I think it's because we disable proto gen validate for Envoy.  Aren't you setting it explicitly to zero while retaining a config setting that it is supposed to be GTE some number of nanos?  Should the restriction be removed entirely or should we not set it to zero for E-M?

Reply:
I changed it to check non-zero before setting the protobuf.